Code/GraphMol/AddHs.cpp

```diff
@@ -275,6 +275,17 @@ void setTerminalAtomCoords(ROMol &mol, unsigned int idx,
                 nbr2Vect =
                     nbr1Pos.directionVector((*cfi)->getAtomPos(nbr2->getIdx()));
                 perpVect = nbr2Vect.crossProduct(nbr1Vect);
+
+                // if nbr1 and nbr2 are aligned, the perpendicular will be null,
+                // so fall back to the default calculation
+                if (perpVect.lengthSq() < sq_dist_zero_tol) {
+                  // we need to pick a different perpendicular
+                  if ((*cfi)->is3D()) {
+                    perpVect = nbr1Vect.getPerpendicular();
+                  } else {
+                    perpVect.z = 1.0;
+                  }
+                }
```

Comment:
Simplification: 
```suggestion
                auto crossProd = nbr2Vect.crossProduct(nbr1Vect);

                // if nbr1 and nbr2 are aligned, the perpendicular will be null,
                // and we'll just keep the default calculated above. Otherwise
                // we use the cross product
                if (crossProd.lengthSq() >= sq_dist_zero_tol) {
                  perpVect = crossProd;
                }
```